pubsub/envelope.go

```diff
@@ -7,36 +7,40 @@ package pubsub
 
 import (
 	"github.com/google/uuid"
-	jsoniter "github.com/json-iterator/go"
 )
 
 const (
 	// DefaultCloudEventType is the default event type for an Dapr published event
 	DefaultCloudEventType = "com.dapr.event.sent"
 	// CloudEventsSpecVersion is the specversion used by Dapr for the cloud events implementation
 	CloudEventsSpecVersion = "1.0"
-	// ContentType is the Cloud Events HTTP content type
-	ContentType = "application/cloudevents+json"
+	// DefaultContentType is the default Cloud Events Envelope content type.
+	DefaultContentType = "application/json"
```

Comment:
![image](https://user-images.githubusercontent.com/32089134/98441444-4e925f00-2139-11eb-990b-0db96d55aef7.png)
https://github.com/cloudevents/spec/blob/master/primer.md#creating-cloudevents
I follow the cloudevent specification in this section, where the implementation is contained directly in a CloudeventEnvelope, regardless of the data being passed.Should we switch to a separate Dapr implementation? Rather than the nesting in the specification.

Reply:
We should not use nesting..

What needs to be done is just the following' if the payload is already a cloud event, then do not create a new envelope but instead continue with the original payload unchanged.

Reply:
This means add logic to handle the origin cloudevent. Do this on 11/12 on schedule.

Reply:
I have a problem with the fact that if we pass existing Cloudevents, we don't know which custom fields the target Cloudevent is carrying, which makes it impossible to parse and can only be encoded as byte slices. Instead of that why don't we embed all the events that need to meet the Cloudevents specification into the Dapr specific Cloud events.
In fact, the origin realization is what you describe. I discuss some here https://github.com/dapr/components-contrib/issues/506
If you decide to use origin realization. I would recover to the previous realization and check what bug he found. Because the issues he found is cloud event stripping the Cloud Event fields.

Reply:
@abserari we (maintainers) discussed your points, what we think should be done is:

When a payload from the user arrives, check if it's a Cloud Event format.
If it is, just pass the existing byte array through. If it's not, do what we did before.

Reply:
Got it. If just pass the existing byte,  the method NewCloudEventEnvelope returns a wrapper just wrap the []byte.

Reply:
sorry for the slow response because of the work of recently. Now the realize just replace the cloudevent.data to cloudevent origin []byte. I think this realization may need more documentation to make users get the right to use.

Reply:
Hey @abserari, actually you can revert back to `m["data"]`, that's the correct behavior. We do not want to nest.

This PR is specifically for allowing users to override the `datatype content` which is great. We will follow up with an additional PR to preserve existing cloud events.

Reply:
ok, I think maybe origin realization would be appropriate. I make code revert back to nothing changed. So is there something in this PR I could do or should I close this PR? Because NewCloudEventEnvelope could override `datatype content`  already.

Reply:
Lets close this PR for now. The changes needed for allowing the Cloud Event envelope to pass as is are more complex than I thought, I need time to check how to do this.